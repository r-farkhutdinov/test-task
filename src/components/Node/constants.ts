export const basePadding = 22;
export const padding = 16;
export const nodeRootTestId = 'node-container';