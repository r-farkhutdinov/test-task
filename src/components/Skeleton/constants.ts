export const defaultLeftPadding = 16;
export const defaultRightPadding = 32;
export const skeletonRootTestId = 'skeleton';