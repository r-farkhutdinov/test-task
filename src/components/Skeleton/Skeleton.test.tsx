import { render, screen } from '@testing-library/react';
import { defaultLeftPadding, defaultRightPadding, skeletonRootTestId } from './constants';
import { Skeleton } from '.';

describe('Skeleton component', () => {
    it('matches snapshot with no props', () => {
        const { asFragment } = render(<Skeleton />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('matches snapshot with level = 2', () => {
        const { asFragment } = render(<Skeleton level={2} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('matches snapshot with isPaddingRight', () => {
        const { asFragment } = render(<Skeleton isPaddingRight />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('matches snapshot with width', () => {
        const { asFragment } = render(<Skeleton width={100} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should apply the correct marginLeft based on level prop', () => {
        const level = 2;
        render(<Skeleton level={level} />);
        expect(screen.getByTestId(skeletonRootTestId)).toHaveStyle(`margin-left: ${level * defaultLeftPadding}px`);
    });

    it('should apply the default marginRight when isPaddingRight=true', () => {
        render(<Skeleton isPaddingRight />);
        expect(screen.getByTestId(skeletonRootTestId)).toHaveStyle(`margin-right: ${defaultRightPadding}px`);
    });

    it('should set the correct width provided', () => {
        const width = 100;
        render(<Skeleton width={width} />);
        expect(screen.getByTestId(skeletonRootTestId)).toHaveStyle(`width: ${width}px`);
    });
});
