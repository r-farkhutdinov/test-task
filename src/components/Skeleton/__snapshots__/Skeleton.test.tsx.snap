// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton component matches snapshot with isPaddingRight 1`] = `
<DocumentFragment>
  <div
    class="root"
    data-testid="skeleton"
    style="margin-left: 0px; margin-right: 32px;"
  />
</DocumentFragment>
`;

exports[`Skeleton component matches snapshot with level = 2 1`] = `
<DocumentFragment>
  <div
    class="root"
    data-testid="skeleton"
    style="margin-left: 32px; margin-right: 0px;"
  />
</DocumentFragment>
`;

exports[`Skeleton component matches snapshot with no props 1`] = `
<DocumentFragment>
  <div
    class="root"
    data-testid="skeleton"
    style="margin-left: 0px; margin-right: 0px;"
  />
</DocumentFragment>
`;

exports[`Skeleton component matches snapshot with width 1`] = `
<DocumentFragment>
  <div
    class="root"
    data-testid="skeleton"
    style="margin-left: 0px; margin-right: 0px; width: 100px;"
  />
</DocumentFragment>
`;
